packages/ui-react/src/components/ProfilesMenu/ProfilesMenu.tsx

```diff
@@ -0,0 +1,69 @@
+import type { UseMutateFunction } from 'react-query';
+import Plus from '@jwp/ott-theme/assets/icons/plus.svg?react';
+import type { Profile } from '@jwp/ott-common/types/profiles';
+import { useTranslation } from 'react-i18next';
+import { useNavigate } from 'react-router-dom';
+import { PATH_USER_PROFILES_CREATE } from '@jwp/ott-common/src/paths';
+
+import styles from '../UserMenu/UserMenu.module.scss'; // TODO inherit styling
+import LoadingOverlay from '../LoadingOverlay/LoadingOverlay';
+import MenuButton from '../MenuButton/MenuButton';
+import Icon from '../Icon/Icon';
+import ProfileCircle from '../ProfileCircle/ProfileCircle';
+
+type ProfilesMenuProps = {
+  profiles: Profile[];
+  currentProfile?: Profile | null;
+  small?: boolean;
+  selectingProfile: boolean;
+  selectProfile?: UseMutateFunction<unknown, unknown, { id: string; avatarUrl: string }, unknown>;
+  onButtonClick?: () => void;
+};
+
+const ProfilesMenu = ({ profiles, currentProfile, small, selectingProfile, selectProfile, onButtonClick }: ProfilesMenuProps) => {
+  const { t } = useTranslation('user');
+  const navigate = useNavigate();
+
+  const onProfileSelectClick = (profile: Profile) => {
+    onButtonClick?.();
+    selectProfile?.({ id: profile.id, avatarUrl: profile.avatar_url });
```

Comment:
We can't make this handler required?

Reply:
Sorry Mike, I should've called this minor, I find the 1 hour investment already a lot. Another time, and thanks for your attempt!